lib/govuk_index/indexable_content_sanitiser.rb

```diff
@@ -0,0 +1,48 @@
+module GovukIndex
+  class IndexableContentSanitiser
+    def initialize(payload)
+      @payload = payload
+    end
+
+    def clean
+      return nil if details.nil?
+
+      cleaned_content = details.values.map { |item|
+        strip_html_tags(indexable_content(item))
+      }.compact
+
+      return nil if cleaned_content.empty?
+      cleaned_content.join("\n")
+    end
+
+  private
+
+    attr_reader :payload
+
+    def details
+      @_details ||= payload.dig('details')
```

Comment:
can we move this into the initializer as we don't actually need the payload in this class (we could actually just pass in the details object).

Also we don't  need to use dig here as we are only getting one level of data.